lib/song_utils/song_normalizer.rb

```diff
@@ -0,0 +1,205 @@
+module SongUtils
+  class BaseParsingStrategy; end
+  class GracepointParsingStrategy < BaseParsingStrategy; end
+  class SongSelectParsingStrategy < BaseParsingStrategy; end
+
+  class SongNormalizer
+    class UnsupportedFileType < StandardError; end
+
+    SUPPORTED_MIME_TYPES = ['application', 'text', 'image']
+    STRATEGIES = [
+      ::SongUtils::GracepointParsingStrategy,
+      ::SongUtils::SongSelectParsingStrategy,
+      ::SongUtils::BaseParsingStrategy
+    ]
+
+    def initialize(path)
+      @garbage_dir = File.expand_path('./tmp/file-conv-dump')
+      @garbage_prefix = 'SNORMALIZER-'
+      Dir.mkdir(@garbage_dir) unless Dir.exists?(@garbage_dir)
+
+      raw_contents = text_for_file(File.expand_path(path))
+      @strategy = parsing_strategy(path, raw_contents)
+      @strategy.execute!
+
+      cleanup_garbage
```

Comment:
Some of the PDFs need to be converted to image formats before the OCR lib (tesseract) can handle them. This method is for removing that temp image that gets created.